core/iwasm/common/wasm_runtime_common.c

```diff
@@ -7322,6 +7322,15 @@ wasm_runtime_sub_module_instantiate(WASMModuleCommon *module,
             (WASMModuleInstance *)sub_module_inst;
         sub_module_inst_list_node->module_name =
             sub_module_list_node->module_name;
+
+#if WASM_ENABLE_AOT != 0
+        if (!init_import_func_module_insts(
```

Comment:
There may be many sub modules, and the function may be called multiple times and import_func_module_insts will be allocated many times?

Reply:
Indeed, I think it would be better to revert to the previous implementation which initializes the `import_func_module_insts` in function `init_func_ptrs`. In this way, we wouldn't need to worry about repeated calls and multiple memory allocations.  What do you think?

Reply:
Maybe you can create array import_func_module_insts before wasm_runtime_sub_module_instantiate:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/867dbd8912f8fd489ea9b24497abb9d3c81f1b40/core/iwasm/aot/aot_runtime.c#L1641-L1643

and then initialize the array elements in wasm_runtime_sub_module_instantiate? If it cannot improve the performance, initializing the array after wasm_runtime_sub_module_instantiate is also good.